contracts/IterableMap.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.3;
+
+import "./IterableMappingAddressUint.sol";
+
+contract IterableMap is IterableMappingAddressUint {
+    mapping (address => uint) data;
+    address[] keys;
+
+    function set(address _key, uint _value) public override
+    {
+        data[_key] = _value;
+        updateKeys(_key);
+    }
```

Comment:
This function works correctly but would be infeasible for use in a real smart contract. Note that every set operation needs to iterate over the `keys` array to check for `_key` uniqueness resulting in a O(n) complexity.

We would expect the get/set/at/length operations to execute in a constant time (O(1)). This can be achieved by storing values in an array and keys in a mapping of `key => arrayIndex`. For a reference implementation see: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.4.0/contracts/utils/EnumerableMap.sol